_posts/2022-12-12-bentoml-deployment-with-zenml.md

```diff
@@ -0,0 +1,269 @@
+
+**BentoML:** is an open-source framework for machine learning model serving. It can be used to serve and deploy models locally, in a cloud environment, or on a Kubernetes cluster. BentoML standardizes the way that models are deployed to production within an organization, making it easier to manage and maintain them. 
+
+To install BentoML we will actually use ZenML CLI integration command which handles this for us:
+
+```shell
+zenml integration install bentoml
+```
+
+Now that we have both tools installed we want to create a ZenML stack that will define how the pipeline will be run, a stack in ZenML consists of multiple stack components and these can be configured as the following:
+
+```shell
+zenml model-deployer register bentoml_deployer --flavor=bentoml
+zenml stack register local_bentoml_stack \
+  -a default \ #refers to the artifact-store stack component
+  -o default \ #refers to the orchestrator stack component
+  -d bentoml_deployer \ # refers to the model-deployer stack component
+  --set
+```
+
+In order to understand how we can deploy models with ZenML and BentoML we will Pytorch to train a classifier on the fashion-mnist dataset. 
+
+The full example can be found in ZenML repository [here](https://github.com/zenml-io/zenml/tree/main/examples/bentoml_deployment)
+
+For more detailed information about stack components, integration and zenml in general you can find it in the docs [here](https://docs.zenml.io/)
+
+## Training Deployment Pipeline
+
+To make this more easily understandable and cleaner we will be splitting the code into two different pipelines. The first one will be the training pipeline, which consists of the following six steps:
+
+1. **Load the Fashion-MNIST dataset**: This step loads the Fashion-MNIST dataset, which is a popular dataset for training image classification models.
+2. **Train a classifier**: In this step, we train a classifier on the Fashion-MNIST dataset using a machine learning algorithm of our choice. This could be a simple model like logistic regression or a more complex model like a convolutional neural network (CNN).
+3. **Evaluator**: After training the classifier, we use the Evaluator step to evaluate the performance of the model on a held-out test set. This allows us to assess the accuracy, precision, and other metrics of our model and helps us determine whether it is ready for deployment.
+4. **Deployment_trigger**: The deployment_trigger step is used to specify when and how the model should be deployed. This step allows us to set certain conditions, such as the performance of the model on the test set, that must be met before the model is deployed.
+5. **Bento_builder**: The bento_builder step takes the trained model and packages it into a format that can be easily deployed and served. This step is important because it allows us to easily share and deploy our trained model to different environments, such as a local machine or the cloud.
```

Comment:
```suggestion
5. **Bento Builder**: The `bento_builder` step converts the trained model and packages it into a format that can be easily deployed and served. This step is important because it allows us to easily share and deploy our trained model to different environments, such as a local machine or the cloud.
```